common/collection/ByteArray.java

```diff
@@ -155,7 +155,7 @@ public static ByteArray encodeInt(int num) {
 
     public abstract int decodeInt();
 
-    public static ByteArray.Base encodeUUID(UUID uuid) {
```

Comment:
the static methods should not be returning the sub (ie. implementation-specific) type, only the more generic `ByteArray`